osc/commands_git/repo_init.py

```diff
@@ -0,0 +1,139 @@
+import sys
+
+import osc.commandline_git
+
+
+class RepoInitCommand(osc.commandline_git.GitObsCommand):
+    """
+    Init a git repo
+
+    A wrapper to git-init, plus merge .gitignore and .gitattributes from a template, plus initialize .git/config from a template
+
+    """
+
+    name = "init"
+    parent = "RepoCommand"
+
+    def init_arguments(self):
+        self.add_argument(
+            "-a",
+            "--anonymous",
+            action="store_true",
+            default=None,
+            help="Clone anonymously via the http protocol",
+        )
+
+        self.add_argument(
+            "path",
+            default=".",
+            help="Destination path to a directory which needs to be initialized. Default is current working directory",
+        )
+
+        self.add_argument(
+            "-t",
+            "--template",
+            default=None,
+            help="Path to a template directory url to git repository",
+        )
+
+        self.add_argument(
+            "-i",
+            "--ssh-key",
+            help="Path to a private SSH key (identity file)",
+        )
+
+        self.add_argument(
+            "--no-ssh-strict-host-key-checking",
+            action="store_true",
+            help="Set 'StrictHostKeyChecking no' ssh option",
+        )
+
+    def run(self, args):
+        import os
+        import subprocess
+        from pathlib import Path
+        from osc.output import tty
+        from osc import gitea_api
+        from osc import git_scm
+
+        template = args.template
+        if not template:
+            template = self.gitea_conf.git_obs_repo_init_template()
+
+        if not template:
+            print("Template must be a valid path or url", file=sys.stderr)
+            exit(1)
+
+        dest = args.path
+        if not dest:
+            dest = "."
+
+        if not dest:
+            print(f"Destination folder ({dest}) must be a valid path", file=sys.stderr)
+            exit(1)
+
+        if not os.path.isdir(dest):
+            os.mkdir(dest)
+
+        if not os.path.isdir(dest):
+            print(f"Destination folder ({dest}) must be a valid path", file=sys.stderr)
+            exit(1)
+
+        gitea_owner, gitea_repo = (None, None)
+        if "://" in template or (not Path(template).is_dir()):
+            path = template
+            # remove trailing ".git" from path
+            if path.endswith(".git"):
+                path = path[:-4]
+
+            gitea_owner, gitea_repo = path.strip("/").split("/")[-2:]
```

Comment:
We should refactor this to a function, I think the code appears in too many places already.